# Graded assignment 1

Mark Klik & Misja Mikkers


# Packages

We need the package `tidyverse`

In [6]:
library(tidyverse) 

# Assignment 1

First, we generate some random data:

In [7]:
df1 <- as.data.frame(matrix(runif(20*2), ncol=20) )
head(df1)

V1        V2        V3        V4        V5        V6         V7       
1 0.4833268 0.3188525 0.9982237 0.6737275 0.9615292 0.80686151 0.7085895
2 0.6424695 0.7247963 0.8699471 0.5090252 0.8264283 0.06608214 0.3409442
  V8        V9        V10       V11        V12       V13        V14      
1 0.8589179 0.0914124 0.5487562 0.04257099 0.8593773 0.09209401 0.1995839
2 0.2772886 0.4233305 0.3409688 0.96747068 0.9590449 0.86969771 0.6093590
  V15       V16       V17       V18       V19       V20      
1 0.1526380 0.4258645 0.7584287 0.7275807 0.2188704 0.1918691
2 0.1892025 0.2973590 0.9541795 0.7366474 0.8595423 0.9542427

_df1_ is a wide dataframe. The goal of the assignment is to change _df1_ into a long dataframe _df2_. And show the first rows of the dataframe using `head(df2)`

The dataframe _df2_ should look like this:

| Key | Value      |
|-----|------------|
| V1  | 0.09413864 |
| V1  | 0.69782526 |
| V2  | 0.23460454 |
| V2  | 0.51520514 |
| V3  | 0.63181844 |
| V3  | 0.22807211 |

In [8]:
df2 <- df1 %>%
  gather()

head(df2)

key value    
1 V1  0.4833268
2 V1  0.6424695
3 V2  0.3188525
4 V2  0.7247963
5 V3  0.9982237
6 V3  0.8699471

## Assignment 2

We will generate some data:

In [9]:
df3 <- as.data.frame(read.table(header=TRUE, text='
      id        V_1       V_2                
          1  42.467573  -87.810047     
          2  42.049148  -88.273029
          3  39.110539  -90.324080
                       '))

head(df3)

id V_1      V_2      
1 1  42.46757 -87.81005
2 2  42.04915 -88.27303
3 3  39.11054 -90.32408

In [10]:
df4 <- as.data.frame(read.table(header=TRUE, text='
     V_3            V_4         id                 
      one         1h            1
      two         1h            1
      three       8h            2
      four        1h            4
'))

head(df4)

V_3   V_4 id
1 one   1h  1 
2 two   1h  1 
3 three 8h  2 
4 four  1h  4


You will have to merge these files, returning all rows and all columns from both _df3_ and _df5_. Where there are not matching values, return _NA_ for the missings. 

This should be the end result:

| id | V_1      | V_2       | V_3   | V_4 |
|----|----------|-----------|-------|-----|
| 1  | 42.46757 | -87.81005 | one   | 1h  |
| 1  | 42.46757 | -87.81005 | two   | 1h  |
| 2  | 42.04915 | -88.27303 | three | 8h  |
| 3  | 39.11054 | -90.32408 | NA    | NA  |
| 4  | NA       | NA        | four  | 1h  |



In [11]:
df5 <- full_join(df3, df4, by='id')
df5

id V_1      V_2       V_3   V_4
1 1  42.46757 -87.81005 one   1h 
2 1  42.46757 -87.81005 two   1h 
3 2  42.04915 -88.27303 three 8h 
4 3  39.11054 -90.32408 NA    NA 
5 4        NA        NA four  1h

End of notebook